Tryouts/Core/BasicModels/MorganStanley.ComposeUI.Tryouts.Core.BasicModels.csproj

```diff
@@ -18,9 +18,16 @@ See the License for the specific language governing permissions and limitations
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="NP.Avalonia.UniDockService" Version="0.99.9" />
```

Comment:
Why are these packages referenced in Core/BasicModels? To me it very much seems like something specific to the shell implementation

Reply:
BasicModels project by its nature contains the non-visual ViewModels to support the visual projects - the shell (the most important visual project), visual plugins and the prototypes. So, yes, this code still needed to be shared between the prototypes and the shell and perhaps (in the future) some plugins also, so I do think that it belongs here. 

UniDockService, BTW is a very simple - **completely non-visual** interface to UniDock (in case you were worried that it has some visual references).

Reply:
No, neither of the packages have any references to Avalonia and they are not Avalonia specific. Nothing is imposed. 

The behaivors - contain non-visual behaviors - that simplify dealing with observable collections. They allow to specify what will happen when an item becomes part of the collection and when it is removed. If you insist I can remove the reference to behaviors package, but it will considerably expand the code. I'd have to add various handlers to the CollectionChanged event figure out the type of the event (adding, removing etc) and do the corresponding action depending on its type - something the behaviors do.

Reply:
I know that we're discussing naming again but why is it called NP.**Avalonia.**UniDockService if it's not Avalonia-specific?

Reply:
+ Something called UniDockService also sounds pretty technology specific to me

Reply:
You are absolutely correct. I'll remove Avalonia from the name - it does not belong there. In case UniDock will be implemented by another framework, e.g. WPF, the UniDockService will still be able to be use for it without a single change. Can we still approve it for now and I'll create a Jira - to use the renamed package for it, once it will be available - next Monday.

Reply:
But this makes a shell that doesn't want to use UniDock dependent on UniDock specific library
If you think a library that provides generic tools for UniDock, move it to a separate project that's thematic to that piece of technology

Reply:
UniDock provides non-visual interface to UniDock. But it does not force anyone to use UniDock. It is a very small set of non-visual code that will not be doing anything back if unidock is not used.

Reply:
adding a project to the reusuable statically loaded project should not be done lightly - it is a major architectural change. What I can probably do - seeing so many objections - I can create a dynamically loaded plugin and use it for the shell.

Reply:
In order to avoid making BasicModels dependent on UniDockServices and Behaviors packages, I moved most of the classes out of there into a new dynamically loaded plugin, leaving there only simple data and interfaces. BasicModels now do not have references to the libraries (those reference were moved to the new plugin - which is no a shared project and can be easily replaced if needed).  
Please, see if you can approve and merge the project. Thanks